src/Tribe/Utils/Element_Classes.php

```diff
@@ -0,0 +1,238 @@
+<?php
+namespace Tribe\Utils;
+
+/**
+ * Class Element_Classes to handle HTML class attribute for elements.
+ *
+ * @since  TBD
+ *
+ * @package Tribe\Utils
+ */
+class Element_Classes {
+	/**
+	 * Store the results of parsing the classes.
+	 *
+	 * @since  TBD
+	 *
+	 * @var array
+	 */
+	protected $result = [];
+
+	/**
+	 * Stores the arguments passed.
+	 *
+	 * @since  TBD
+	 *
+	 * @var array
+	 */
+	protected $arguments = [];
+
+	/**
+	 * Setups an instance of Element Classes.
+	 *
+	 * @since  TBD
+	 *
+	 * @return void
+	 */
+	public function __construct() {
+		$this->arguments = func_get_args();
+	}
+
+	/**
+	 * When invoked this class will return the full HTML class attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function __invoke() {
+		$this->arguments = func_get_args();
+		return $this->get_attribute();
+	}
+
+
+	/**
+	 * When cast to string an instance will return the full HTML class attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function __toString() {
+		return $this->get_attribute();
+	}
+
+	/**
+	 * Gets the full HTML class attribute for this instance of Element Classes.
+	 * It will contain a space on each end of the attribute.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string In the format ` class="class1 class2" `
+	 */
+	public function get_attribute() {
+		$classes = $this->get_classes_as_string();
+		return " class=\"{$classes}\" ";
+	}
+
+	/**
+	 * Gets a space separated string of all classes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return string
+	 */
+	public function get_classes_as_string() {
+		return implode( ' ', $this->get_classes() );
+	}
+
+	/**
+	 * Get the array of classes to be printed.
+	 *
+	 * @since  TBD
+	 *
+	 * @return array
+	 */
+	public function get_classes() {
+		$this->results = [];
+		$classes       = [];
+
+		$this->parse_array( $this->arguments );
+
+		foreach ( $this->results as $key => $val ) {
+			if ( ! $val ) {
+				continue;
+			}
+
+			$classes[] = $key;
+		}
+
+		$classes = array_map( 'sanitize_html_class', $classes );
+		$classes = array_unique( $classes );
```

Comment:
```suggestion
		$classes = array_filter( array_unique( $classes ) );
```

Might be a nice if the result of sanitizing means an empty string